### PR TITLE
[Block Library - Site Logo]: Return nothing if logo is not set

### PR DESCRIPTION
**Pull request by jffng**
## Description
Currently, the site logo block renders an empty div if the theme has support for a custom logo but the user has not set one. 

This PR adjusts the render callback to return nothing if the custom logo has not been set. 

## How has this been tested?
- Activate a theme with custom logo support
- Remove the logo if one has been set
- Verify that no markup is generated by the block  

## Types of changes
Bug fix

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by mtias — APPROVED** [review]
Might be worth a comment

**Comment by carolinan** [comment]
I was able to reproduce the problem, and the pr solves it.

_Sidenote: Without the customizer it is still not possible to remove the image from the logo block, only replace it or remove the entire block._

File: packages/block-library/src/site-logo/index.php [hunk 1]
```diff
@@ -22,8 +22,13 @@ function render_block_core_site_logo( $attributes ) {
 	};
 
 	add_filter( 'wp_get_attachment_image_src', $adjust_width_height_filter );
+
 	$custom_logo = get_custom_logo();
 
+	if ( empty( $custom_logo ) ) {
+		return ''; // Return early if no custom logo is set, avoiding extraneous wrapper div.
+  }
```
Comment (carolinan) [inline packages/block-library/src/site-logo/index.php]: Sorry Jeff, I broke your code when trying to solve the merge conflict via the GitHub UI 👎 . There is supposed to be a tab here.
```suggestion
	}
```

**Review by ntsekouras — APPROVED** [review]

**Comment by mtias** [comment]
> Without the customizer it is still not possible to remove the image from the logo block, only replace it or remove the entire block.

Added it as a task in #30406 so we remember.

**Pull request merged**